packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -581,14 +597,24 @@ class _DatePickerDialogState extends State<DatePickerDialog> with RestorationMix
           : localizations.datePickerHelpText.toUpperCase()
       ),
       titleText: dateText,
-      titleStyle: dateStyle,
+      titleStyle: headlineStyle,
       orientation: orientation,
       isShort: orientation == Orientation.landscape,
       entryModeButton: entryModeButton,
     );
 
     final Size dialogSize = _dialogSize(context) * textScaleFactor;
+    final DialogTheme dialogTheme = Theme.of(context).dialogTheme;
```

Comment:
Once you have one, you can use the `theme` local variable here.

Reply:
Addressed in https://github.com/flutter/flutter/pull/119033/commits/70d860d720f506a57c730749eaaba252e4cd8a6b